contracts/payment/MCPayment.sol

```diff
@@ -89,21 +106,33 @@ contract MCPayment is Ownable2StepUpgradeable, EIP712Upgradeable {
         $.ownerPercentage = ownerPercentage;
     }
 
-    function pay(
+    function nativeCurrencyPayment(
```

Comment:
`payNativeCurrency` and `payERC20Token` may be somewhat better

Reply:
sure, renamed

Reply:
vote just for 
- pay ( we omit word native if it is native in protocol)
- payERC20  (omit word token)

Reply:
Yes, I agree with @vmidyllic proposal for payment functions names. They look simpler and more clear.

Reply:
renamed